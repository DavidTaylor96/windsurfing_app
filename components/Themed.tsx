import * as React from "react";
import { Text as DefaultText, View as DefaultView } from "react-native";

import Colors from "../constants/Colors";
import useColorScheme from "../hooks/useColorScheme";

export function useThemeColor(
  props: { light?: string; dark?: string },
  colorName: keyof typeof Colors.light & keyof typeof Colors.dark
) {
  const theme = useColorScheme();
  const colorFromProps = props[theme];

  if (colorFromProps) {
    return colorFromProps;
  } else {
    return Colors[theme][colorName];
  }
}

type ThemeProps = {
  lightColor?: string;
  darkColor?: string;
};

type FontWeightProps = {
  weight?: "normal" | "h1" | "h2" | "h3" | "h4" | "h5";
};

export type TextProps = FontWeightProps & ThemeProps & DefaultText["props"];
export type ViewProps = ThemeProps & DefaultView["props"];

export function Text(props: TextProps) {
  const { style, lightColor, darkColor, ...otherProps } = props;
  const color = useThemeColor({ light: lightColor, dark: darkColor }, "text");

  let fontSize = 14;
  let fontFamily:
    | "sora-bold"
    | "sora-extra-bold"
    | "sora-extra-light"
    | "sora-light"
    | "sora-medium"
    | "sora-regular"
    | "sora-semi-bold"
    | "sora-thin" = "sora-regular";

  switch (props.weight) {
    case "normal":
      fontSize = 12;
      break;
    case "h1":
      fontSize = 24;
      break;
    case "h2":
      fontSize = 24;
      fontFamily = "sora-semi-bold";
      break;
    case "h3":
      fontSize = 18;
      break;
    case "h4":
      fontSize = 16;
      break;
    case "h5":
      fontSize = 16;
      fontFamily = "sora-semi-bold";

      break;
    default:
      fontSize = 12;
  }

  return (
    <DefaultText
      style={[{ fontSize, fontFamily }, { color }, style]}
      {...otherProps}
    />
  );
}

export function View(props: ViewProps) {
  const { style, lightColor, darkColor, ...otherProps } = props;
  const backgroundColor = useThemeColor(
    { light: lightColor, dark: darkColor },
    "background"
  );

  return <DefaultView style={style} {...otherProps} />;
}
