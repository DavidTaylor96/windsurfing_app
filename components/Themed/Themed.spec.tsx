import * as React from 'react';
import renderer from 'react-test-renderer';
import { Text, View } from './Themed';

it(`Text H2 test`, () => {
  const tree = renderer
    .create(<Text weight="h2">Snapshot test!</Text>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
it(`Text H1 test`, () => {
  const tree = renderer
    .create(<Text weight="h1">Snapshot test!</Text>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
it(`Text H3 test`, () => {
  const tree = renderer
    .create(<Text weight="h3">Snapshot test!</Text>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
it(`Text H4 test`, () => {
  const tree = renderer
    .create(<Text weight="h4">Snapshot test!</Text>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
it(`Text H5 test`, () => {
  const tree = renderer
    .create(<Text weight="h5">Snapshot test!</Text>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
it(`Text P test`, () => {
  const tree = renderer
    .create(<Text>Snapshot test!</Text>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});


it(`View`, () => {
  const view = renderer.create(<View />)
  expect(view).toMatchSnapshot();
})
