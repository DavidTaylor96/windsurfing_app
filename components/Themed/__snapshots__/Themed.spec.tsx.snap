// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text H1 test 1`] = `
<Text
  style={
    Array [
      Object {
        "fontFamily": "sora-regular",
        "fontSize": 24,
      },
      Object {
        "color": "black",
      },
      undefined,
    ]
  }
  weight="h1"
>
  Snapshot test!
</Text>
`;

exports[`Text H2 test 1`] = `
<Text
  style={
    Array [
      Object {
        "fontFamily": "sora-semi-bold",
        "fontSize": 24,
      },
      Object {
        "color": "black",
      },
      undefined,
    ]
  }
  weight="h2"
>
  Snapshot test!
</Text>
`;

exports[`Text H3 test 1`] = `
<Text
  style={
    Array [
      Object {
        "fontFamily": "sora-regular",
        "fontSize": 18,
      },
      Object {
        "color": "black",
      },
      undefined,
    ]
  }
  weight="h3"
>
  Snapshot test!
</Text>
`;

exports[`Text H4 test 1`] = `
<Text
  style={
    Array [
      Object {
        "fontFamily": "sora-regular",
        "fontSize": 16,
      },
      Object {
        "color": "black",
      },
      undefined,
    ]
  }
  weight="h4"
>
  Snapshot test!
</Text>
`;

exports[`Text H5 test 1`] = `
<Text
  style={
    Array [
      Object {
        "fontFamily": "sora-semi-bold",
        "fontSize": 16,
      },
      Object {
        "color": "black",
      },
      undefined,
    ]
  }
  weight="h5"
>
  Snapshot test!
</Text>
`;

exports[`Text P test 1`] = `
<Text
  style={
    Array [
      Object {
        "fontFamily": "sora-regular",
        "fontSize": 12,
      },
      Object {
        "color": "black",
      },
      undefined,
    ]
  }
>
  Snapshot test!
</Text>
`;

exports[`View 1`] = `<View />`;
